#168360958 create signup endpoint with persistence

jo-rdan opened the pull request:
#### What does this PR do?
It creates an endpoint for signing up with persistence to the database
#### Description of Task to be completed?
- create signup endpoint with persistence to database
- write unit tests for signup endpoint to test if the endpoint works as expected
#### How should this be manually tested?
- Clone this repository to your local computer
- Open the folder you cloned and start the server by typing **npm start**
- start **POSTMAN** and use this endpoint **/api/v2/auth/signup** and enter all fields correctly and it should return a **201** HTTP status code, otherwise it will respond with a relevant error message 
#### Any background context you want to provide?
- N/A
#### What are the relevant pivotal tracker stories?
[#168360958](https://www.pivotaltracker.com/story/show/168360958)
### Screenshots (if appropriate)
**When everything is fine**
![image](https://user-images.githubusercontent.com/51836855/64545739-25303600-d2de-11e9-8c3e-959751804cc0.png)


**When error**
![image](https://user-images.githubusercontent.com/51836855/64545668-05007700-d2de-11e9-8d68-5849b087bce4.png)




#### Questions:
- N/A

jo-rdan committed:
feat(signup): create signup endpoint
- create signup endpoint with persistence to database
- write unit tests for signup endpoint
[Finishes #168360958]

jo-rdan committed:
feat(signup): create signup endpoint
- create signup endpoint with persistence to database
- write unit tests for signup endpoint
[Finishes #168360958]

Oluwasegun-AA commented:
![image](https://user-images.githubusercontent.com/25525765/64539324-97c10600-d31e-11e9-8209-d2b527bfaa4a.png)
please refactor your error response. it should be 
```
{
"status": "integer",
"error": "string"
}
```

Oluwasegun-AA commented on server/app.js:
```diff
@@ -7,13 +7,15 @@ import bodyparser from 'body-parser';
-app.use('/api/v1/auth/', userRoute);
-app.use('/api/v1/', userRoute);
-app.use('/api/v1/', adminRoute);
+app.use('/api/v2/auth/', userRoute);
```
@jo-rdan, please make your routes modular. kindly implement the modularity of your routes in this PR

Oluwasegun-AA commented on server/controllers/user_controller.js:
```diff
@@ -4,34 +4,43 @@ import bcrypt from 'bcrypt';
-      const token = jwt.sign({ id:newUser.menteeId, email:newUser.email, isAdmin:newUser.isAdmin }, process.env.secret);
-      return res.status(201).send({ status: 201, message: 'User created successfully', data: {token}});
+      ]
+      // User.create(newUser);
```
@jo-rdan please remove all redundant/ commented lines of  codes

Oluwasegun-AA commented on server/helpers/userServer.js:
```diff
@@ -1,14 +1,17 @@
 /* eslint-disable */
```
@jo-rdan  do not disable any Eslint flag. the purpose of using the Airbnb configuration is to ensure that we write our code in line with a predefined standard. kindly remove all Eslint disabling comments and adjust the codes accordingly

jo-rdan committed:
feat(signup): create signup endpoint
- create signup endpoint with persistence to database
- write unit tests for signup endpoint
[Finishes #168360958]

jo-rdan committed:
feat(signup): create signup endpoint
- create signup endpoint with persistence to database
- write unit tests for signup endpoint
- create index file in routes to handle routes
[Finishes #168360958]

jo-rdan committed:
feat(signup): create signup endpoint
- create signup endpoint with persistence
- write unit tests for the endpoint
- remove eslint disabling comments on all files
[Finishes #168360958]

jo-rdan committed:
feat(signup): create signup endpoint
- create signup endpoint with persistence
- write unit tests for the endpoint
- remove eslint disabling comments on all files
[Finishes #168360958]
